### ISSUE TITLE
BUG: Badge.query can return duplicate indices within a single batch

### ISSUE THREAD

**Issue by TioSisai** [scikit-activeml/scikit-activeml#586]
### Describe the issue:

`Badge.query(..., batch_size=6)` can return the same index twice, so the batch
contains fewer than `batch_size` distinct samples and the AL loop silently gets
fewer labels than it paid for.

Expected 6 distinct indices, got `[2, 9, 8, 9, 6, 7]` — index 9 appears twice.

Versions: `scikit-activeml == 1.0.0` (current `master`), `numpy == 2.4.6`,
`scikit-learn == 1.9.0`.

### Reproduce the bug via a code example:

```python
import numpy as np
from sklearn.linear_model import LogisticRegression
from skactiveml.classifier import SklearnClassifier
from skactiveml.pool import Badge
from skactiveml.utils import MISSING_LABEL

# 12 samples: 0 and 1 are labeled, the 10 candidates come from two repeated positions.
X = np.vstack([np.array([[0.0, 0.0], [1.0, 1.0]]),
               np.tile(np.array([[3.0, 3.0], [7.0, 7.0]]), (5, 1))])
y = np.hstack([[0, 1], np.full(10, MISSING_LABEL)])
clf = SklearnClassifier(LogisticRegression(), classes=[0, 1])

query_indices = Badge(random_state=5).query(X, y, clf, batch_size=6)
print("query_indices =", np.asarray(query_indices).tolist())
print("n_distinct    =", len(set(np.asarray(query_indices).tolist())))
```

### Error message:

```shell
No traceback, the malformed batch is returned silently:


query_indices = [2, 9, 8, 9, 6, 7]
n_distinct    = 5
```

### Context for the issue:

A duplicate means the batch is short: Ask for 6 labels and got 5 samples to
annotate, with no warning. Over an AL loop this quietly changes how much of the
labeling budget each round actually spends, and an annotation pipeline that keys
on the sample index sees the same sample twice.
